packages/mui-material/src/Button/Button.js

```diff
@@ -345,6 +360,12 @@ const Button = React.forwardRef(function Button(inProps, ref) {
     </ButtonEndIcon>
   );
 
+  let extendedOtherProp = other;
+  if (buttonGroupButtonContext) {
+    const dataAttrs = addPositionDataAttr(buttonGroupButtonContext);
+    extendedOtherProp = { ...dataAttrs, ...other };
+  }
```

Comment:
Why add `dataAttrs` to the `other` props instead of adding it directly?

Reply:
By adding it directly, do you mean adding it in the returned JSX below? Like:
``` jsx
let dataAttrs = {}
if (buttonGroupButtonContext) {
  dataAttrs = addPositionDataAttr(buttonGroupButtonContext);
}

return <ButtonRoot ...  {...dataAttrs} {...other} />;
```

Reply:
Yes

Reply:
I guess it's just a matter of readability. I've applied your suggestion.